contracts/proposal/dao-proposal-single/src/contract.rs

```diff
@@ -255,6 +258,77 @@ pub fn execute_propose(
         .add_attribute("status", proposal.status.to_string()))
 }
 
+pub fn execute_veto(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    proposal_id: u64,
+) -> Result<Response, ContractError> {
+    let mut prop = PROPOSALS
+        .may_load(deps.storage, proposal_id)?
```

Comment:
What's is the reason for `may_load` instead of `load` here? Should be better just end here if no proposal found no?

Reply:
Here is the a snippet which works for me. But I see the `may_load` pattern is the standard in the source code in the other places.

`let mut prop = PROPOSALS
        .load(deps.storage, proposal_id)
        .or(Err(ContractError::NoSuchProposal { id: proposal_id }))?;`